docs/datastorage/capabilities.md

```diff
@@ -0,0 +1,176 @@
+
+Forge adds capability support to TileEntities, Entities, and ItemStacks, which can be exposed either by attaching them through an event or, with the exception of ItemStacks, by overriding he capability methods in your own implementations of the entities. This will be explained in more detail in the following sections.
+
+Using an Existing Capability
+----------------------------
+
+This most notably applies to the Item Handler capability added by Forge, which replaces the vanilla IInventory and ISidedInventory with an automation-friendly system.
+
+As mentioned earlier, TileEntities, Entities, and ItemStacks implement the capability provider feature, through the `ICapabilityProvider` interface. This interface adds two methods, `hasCapability` and `getCapability`, which can be used to query the capabilities present in the objects.
+
+In order to obtain a capability, you will need to refer it by its unique instance. In the case of the Item Handler, this capability is primarily stored in `CapabilityItemHandler.ITEM_HANDLER_CAPABILITY`, but it is possible to get an instance reference by using the `@CapabilityInject` annotation. This annotation can also be used on a method in order to get notified when a capability is registered, so that certain features can be enabled conditionally. Please refer to the javadocs for this annotation for details on usage.
+
+Both the `hasCapability` and `getCapability` methods have a second parameter, of type EnumFacing, which can be used in the TileEntities to request the specific instance for that one face. If passed `null`, a general capability instance will be requested instead. The return type of `getCapability` will correspond to the type declared in the capability passed to the method. For the Item Handler capability, this is indeed `IItemHandler`.
+
+Exposing a Capability
+---------------------
+
+In order to expose a capability, you will first need an instance of the underlying capability type. There's two ways to obtain such an instance, through the Capability itself, or by explicitly instantiating an implementation of it.
+
+The first method is designed to use a default implementation, if those default values are useful for you. In the case of the Item Handler capability, the default implementation will expose a single slot inventory, which is most probably not what you want.
+
+The second method can be used to customize the existing implementations, or to implement your own. In the case of `IItemHandler`, the default implementation uses the `ItemStackHandler` class, which has an optional argument in the constructor, to specify a number of slots.
+
+```Java
+// A dispenser-like block could use something like this
+IItemHandler inventory = new ItemStackHandler(9);
+```
```

Comment:
It is important to note NOT to do this in static fields/methods. AND gate them behind a if (CAPABILITY_HAS_BEEN_REGISTERED) {Typically if (Field_With_CapInject != null)}
As the main point of this is to prevent class not found exceptions when you are in an environment that the capability doesn't exist.